ui/packages/components/src/Timeline/TimelineNode/TimelineNode.tsx

```diff
@@ -107,20 +127,52 @@ function Content({
     <>
       <div className="pb-5">
         <MetadataGrid
-          metadataItems={[
-            {
-              label: 'Started At',
-              value: node.scheduledAt ? node.scheduledAt.toLocaleString() : '-',
-            },
-            {
-              label: 'Ended At',
-              value: node.endedAt ? node.endedAt.toLocaleString() : '-',
-            },
-            {
-              label: 'Duration',
-              value: durationMS ? formatMilliseconds(durationMS) : '-',
-            },
-          ]}
+          metadataItems={
+            [
+              {
+                label: 'Started At',
+                value: node.scheduledAt ? node.scheduledAt.toLocaleString() : '-',
+                title: node?.scheduledAt?.toLocaleString(),
+              },
+              {
+                label: endedAtLabel,
+                value: node.endedAt ? node.endedAt.toLocaleString() : '-',
+                title: node?.endedAt?.toLocaleString(),
+              },
+              {
+                label: 'Duration',
+                value: durationMS ? formatMilliseconds(durationMS) : '-',
+                tooltip: `Time between ${type ? type : 'step'} started and ${
+                  type ? type : 'step'
+                } ${tootltipLabel}`,
+              },
+              ...(node.sleepConfig?.until
```

Comment:
Building `metadataItems` in this way is kinda tough to read now and will get tougher as we support more step types. I think the ternaries and deep nesting (13 indents at one point) are the biggest issues for me